src/routes/auth/index.ts

```diff
@@ -0,0 +1,147 @@
+	} catch (error) {
+		console.error('Error creating profile', error);
+		return context.json<StatusResponse>({ message: 'Error creating profile' }, StatusCodes.INTERNAL_SERVER_ERROR);
+	}
+
+	// Send email
+	try {
+		sgMail.setApiKey(context.env.SENDGRID_API_KEY);
+
+		const salt = crypto.getRandomValues(new Uint8Array(16));
+		const token = await hashPasswordPBKDF2(parsedBody.email, salt);
+		const encodedToken = encodeURIComponent(token);
+		await context.env.ACTIVATION_TOKENS.put(
+			encodedToken,
+			parsedBody.email,
+			{ expirationTtl: 60 * 10 }
+		);
+
+		const emailHtmlTemplate = generateEmailHtml(context, encodedToken);
+		const msg = {
+			to: parsedBody.email,
+			from: context.env.SENDER_EMAIL,
+			subject: '[TorontoJS] Confirm your account',
+			html: emailHtmlTemplate
+		};
+		await sgMail.send(msg);
+
+		return context.json<StatusResponse>({ message: 'Sent an email successfully' }, StatusCodes.OKAY);
+	} catch (error) {
+		console.error('Error sending email', error);
+		return context.json<StatusResponse>({ message: 'Error sending email' }, StatusCodes.INTERNAL_SERVER_ERROR);
+	}
+});
+
+authRoutes.get('/activate', async (context: Context<EnvironmentBindings>) => {
+	try {
+		const token = context.req.query('token');
+		if (!token) {
+			return context.json({ message: 'Invalid or missing token' }, StatusCodes.BAD_REQUEST);
+		}
+
+		const email = await context.env.ACTIVATION_TOKENS.get(token);
+		if (!email) {
+			return context.json({ message: 'Invalid or expired token' }, StatusCodes.UNAUTHORIZED);
+		}
+
+		const emailExists = await checkEmail(context.env.database, email);
+		if (!emailExists) {
+			return context.json({ message: 'User not found' }, StatusCodes.NOT_FOUND);
+		}
+
+		await activateProfile(context.env.database, email);
+
+		// Remove token after successful activation
+		await context.env.ACTIVATION_TOKENS.delete(token);
+
+		return context.json({ message: 'Account activated successfully' }, StatusCodes.OKAY);
+	} catch (error) {
+		return context.json<StatusResponse>({ message: 'Error activating an account' }, StatusCodes.INTERNAL_SERVER_ERROR);
+	}
+});
+
+authRoutes.post('/sign-in', async (context: Context<EnvironmentBindings>) => {
+	try {
+		const body = await context.req.json();
+		const parsedBody = SignInSchema.parse(body);
+
+		const storedPassword = await getPassword(context.env.database, parsedBody);
+
+		if (!storedPassword) {
+			return context.json<StatusResponse>({ message: 'Unauthorized requests' }, StatusCodes.UNAUTHORIZED);
+		}
+
+		// Extract salt and hashed password from stored format
+		const [saltBase64, storedHashedPassword] = storedPassword.split(':');
+		if (saltBase64 === undefined) {
+			throw new Error('Salt not found');
+		}
+		const salt = new Uint8Array([...atob(saltBase64)].map((c) => c.charCodeAt(0)));
+
+		// Hash input password with the same salt
+		const inputHashedPassword = await hashPasswordPBKDF2(parsedBody.password, salt);
```

Comment:
Move this to a password utility function too. I'd basically copy the `bcrypt` signature and expose 2 methods: `hashPassword` and `validatePassword`.